modules/apigee_edge_apidocs/src/ApiDocAccessControlHandler.php

```diff
@@ -59,6 +82,63 @@ protected function checkAccess(EntityInterface $entity, $operation, AccountInter
     return $parent_access;
   }
 
+  /**
+   * Additional access control for revisions.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity for which to check access.
+   * @param string $operation
+   *   The entity operation.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The user for which to check access.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected function checkAccessRevisions(EntityInterface $entity, $operation, AccountInterface $account) {
+    /** @var \Drupal\Core\Entity\EntityStorageInterface $entity_storage */
+    $entity_storage = $this->entityTypeManager->getStorage($this->entityTypeId);
+
+    // Must have access to the same operation on the default revision.
+    $default_revision = $entity_storage->load($entity->id());
+    $entity_access_default = $default_revision->access($operation, $account);
+    if (!$entity_access_default) {
+      return AccessResult::forbidden();
+    }
+
+    $map = [
+      'view' => "view all {$this->entityTypeId} revisions",
```

Comment:
I think you are unintentionally mixing what is the scope that an entity access handler should handle and what is the scope of a route access handler.
https://github.com/fago/entity/blob/85f76b56f25d7c3f5bc466f94db30a9b63d97fc5/src/Access/EntityRevisionRouteAccessChecker.php#L86 This is a route access handler and you can see the similarity between your implementation and this class.

Reply:
So probably this is the right place to ask again, can we get rid of the custom entity access and entity route handlers that we implemented and rely on what the Entity module provides? 
I would feel much safer if access handlers from a well-tested framework would ensure the entity access in our system.

Reply:
@Jaesin @cnovak Please share your thoughts about this ^ We are still maintaining and testing our own solution instead of relying on an OOTB solution by Entity module - which has its own test coverage.

Reply:
Yet an other reason why the custom implementation should go:
https://github.com/apigee/apigee-edge-drupal/pull/161/files/901a1e5c817a9447528c55d1cfe1524151f7e7ad#r274771293